server/plugin.go

```diff
@@ -72,6 +73,15 @@ type Plugin struct {
 
 	WebhookHandler webhook.Webhook
 	GitlabClient   gitlab.Gitlab
+	// gitlabPermalinkRegex is used to parse gitlab permalinks in post messages.
+	gitlabPermalinkRegex *regexp.Regexp
+}
+
+// NewPlugin returns an instance of a Plugin.
+func NewPlugin() *Plugin {
+	return &Plugin{
+		gitlabPermalinkRegex: regexp.MustCompile(`https?://(?P<haswww>www\.)?gitlab\.com/(?P<user>[\w/.?-]+)/(?P<repo>[\w-]+)/-/blob/(?P<commit>\w+)/(?P<path>[\w-/.]+)#(?P<line>[\w-]+)?`),
```

Comment:
Have you checked if there's a difference between Github and Gitlab permalinks or just copied this regex from Github plugin?

Reply:
I have copied it from Github plugin and then made changes according to the GitLab plguin.